cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/aggregation/ContextRetriever.java

```diff
@@ -111,38 +111,40 @@ public ContextRetriever(Map<String, String> inputPaths, DatasetRetriever dataset
             String relatedDataType = join.getRelatedDataType();
             String relatedColumnName = join.getRelatedKeyColumn();
             Dataset<Row> relatedDataset = readDataset(relatedDataType);
-            Dataset<Row> joinedDataset;
-
-            if (join.getClass() == OneToMany.class) {
-                Column joinCriteria = primaryDataset.col(primaryJoinColumn)
-                        .equalTo(relatedDataset.col(relatedColumnName));
-                joinedDataset = primaryDataset.join(relatedDataset, joinCriteria);
-            } else if (join.getClass() == ManyToMany.class) {
-                ManyToMany manyToMany = (ManyToMany) join;
-                String assocDataType = manyToMany.getAssociationDataType();
-                Dataset<Row> assocDataset = readDataset(assocDataType);
-                String assocPrimaryColumnName = manyToMany.getAssociationOneKeyColumn();
-                String assocRelatedColumnName = manyToMany.getAssociationManyKeyColumn();
-
-                Column primaryJoinCriteria = primaryDataset.col(primaryJoinColumn)
-                        .equalTo(assocDataset.col(assocPrimaryColumnName));
-                joinedDataset = primaryDataset.join(assocDataset, primaryJoinCriteria);
-
-                Column relatedJoinCriteria = joinedDataset.col(assocRelatedColumnName)
-                        .equalTo(relatedDataset.col(relatedColumnName));
-                joinedDataset = joinedDataset.join(relatedDataset, relatedJoinCriteria);
-            } else {
-                throw new IllegalArgumentException("Unexpected Join Type: " + join.getClass().getName());
+            if( relatedDataset != null ) {
```

Comment:
I think it's worth logging a warning that a particular datatype join didn't come back with a Dataset.
I _think_ this only happens when a user specifies a join, but never actually references the type in their CQLs.
If an unexpected null ever makes it through, the log could help indicate what happened.

Reply:
The null is new with my change. It was added to support exclusion of dataset reads when the data in that dataset is never actually used in the CQL code. I wouldn't log it as a warning, but I have no objection to logging an INFO message here. I agree it could be useful in debugging.